src/ngraph/op/lrn.cpp

```diff
@@ -34,16 +45,35 @@ op::LRN::LRN(const Output<Node>& arg, double alpha, double beta, double bias, si
 
 void op::LRN::validate_and_infer_types()
 {
-    UnaryElementwiseArithmetic::validate_and_infer_types();
+    element::Type arg_type = get_input_element_type(0);
+    PartialShape arg_shape = get_input_partial_shape(0);
+    set_output_type(0, arg_type, arg_shape);
 
     const PartialShape& input_shape = get_input_partial_shape(0);
+    const PartialShape& axes_shape = get_input_partial_shape(1);
 
     NODE_VALIDATION_CHECK(this,
                           input_shape.rank().is_dynamic() ||
                               static_cast<size_t>(input_shape.rank()) >= 3,
                           "Argument must have rank >= 3 (argument shape: ",
                           input_shape,
                           ").");
+
+    NODE_VALIDATION_CHECK(this, axes_shape.is_static(), "Input axes must be static.");
```

Comment:
(Going to go ahead and comment on this but I won't block on this in review. If you don't get a chance to update to address it, please open a JIRA.)

We should be able to handle all cases here:

1. If `axes_shape.is_static()` returns false, we just assume everything is okay. Else...
2. If `axes_shape` is not 1, we throw an exception (your PR is already doing this). Else...
3. If input 1 (the axes input) is not being driven by a constant, we just assume everything is okay. Else...
4. We get the value out of the constant node, and make sure that that value is in bounds.

Reply:
Could you explain what do you mean by `Else...` statement. Should we handle all listed cases? For example adding support for dynamic shapes is required?

Reply:
Supporting dynamic shapes is required. So we have to handle all cases. (Shape of "axes" tensor is dynamic; shape of "axes" tensor is static and invalid; shape of "axes" tensor is static and valid but the value of the "axes" tensor is unknown; value of the "axes" tensor is known but invalid; and value of the "axes" tensor is valid.)

Reply:
Thank for explanation. Because the PR was merged I'll add all these requirements to the JIRA task (created for LRN reference implementation)